package.json

```diff
@@ -1,16 +1,18 @@
 {
         "name": "mako",
         "description": "Storage backend for Manta",
-        "version": "2.0.0",
+        "version": "2.0.1",
         "author": "Joyent (joyent.com)",
         "license": "MPL-2.0",
         "private": true,
         "dependencies": {
+                "@smaller/tap": "11.1.4-1.0.0",
                 "assert-plus": "1.0.0",
                 "bunyan": "1.8.12",
                 "kthxbai": "0.4.0",
+                "node-uuid": "^1.4.8",
                 "restify": "4.3.4",
-                "@smaller/tap": "11.1.4-1.0.0",
+                "tape": "^4.13.2",
```

Comment:
Yikes this adds about 300k (compressed!) to the size of the image. Oh well.

Reply:
huh, I have no recollection of adding that - wtf?

Reply:
It's ok, I'm merely losing my mind. This was needed for us to be able to run the mpu tests from a mako zone, otherwise:
```
[root@9f535ba6 (storage) /opt/smartdc/mako]$ ./build/node/bin/node $(find test/ -type f -name "*.js")
module.js:550
    throw err;
    ^

Error: Cannot find module 'tape'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/opt/smartdc/mako/test/mpu.test.js:15:16)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
```
I'll move it to being a dev dependency and update the Makefile `test:` target advice

Reply:
Ugh, bit of a mess in there, but doing `npm install --production` instead when building the release tarball gets us down from a 40mb node_modules directory to a 16mb one.

The `check:` Makefile target will still be run by Jenkins before a build, bringing in eslint & co, and we `git clean -fdx` those before building the release tarball, so we should be fine there.

For extra paranoia points, I built a mako image with the changes, deployed it in coal, sanity checked manta put/get operations and ran the test suite *without* the mpu tests, to ensure that nothing was actually depending on the devDependencies that we have been inadvertently shipping to date.

Maybe this is overkill and we don't really mind the extra junk in the image. Another option would be to include `tape` as a normal dependency, so that at least all the tests work out of the box. From trying that, on top of the 16mb of packages, adding tape bumps us up to 22mb. Still down from the 40mb we had originally.

I'll go with that last suggestion, re-test just to make sure, and we can discuss.

```
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461) rm -rf node_modules
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461) make all
.
.
.
PATH=/mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin:/opt/local/bin:/opt/local/sbin:/usr/bin:/usr/sbin:/home/timf/bin:/root/bin:/root/opt/node/bin:/home/timf/projects/node_modules/.bin:/opt/local/bin:/opt/local/sbin:/usr/bin:/usr/sbin:/home/timf/bin:/root/opt/node/bin:/home/timf/projects/node_modules/.bin:/usr/ccs/bin:/usr/bin:/bin:/usr/sbin:/sbin:/root/bin:/root/bin:/opt/tools/bin /mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/node /mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/npm install --production
.
.
.
found 5 vulnerabilities (3 low, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461) du -hs node_modules
22M     node_modules
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461) make test
xmllint --noout --path deps/eng/tools/ --dtdvalid deps/eng/tools/service_bundle.dtd.1 smf/manifests/nginx.xml
PATH=/mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin:/opt/local/bin:/opt/local/sbin:/usr/bin:/usr/sbin:/home/timf/bin:/root/bin:/root/opt/node/bin:/home/timf/projects/node_modules/.bin:/opt/local/bin:/opt/local/sbin:/usr/bin:/usr/sbin:/home/timf/bin:/root/opt/node/bin:/home/timf/projects/node_modules/.bin:/usr/ccs/bin:/usr/bin:/bin:/usr/sbin:/sbin:/root/bin:/root/bin:/opt/tools/bin /mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/node /mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/npm install --no-save eslint@^4.19.1 eslint-config-prettier@^6.0.0 eslint-plugin-joyent@^2.1.0 eslint-plugin-prettier@^3.1.0 prettier@^1.18.2 && \
    touch node_modules/.bin/eslint
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
npm WARN eslint-plugin-prettier@3.1.3 requires a peer of eslint@>= 5.0.0 but none is installed. You must install peer dependencies yourself.

+ eslint-config-prettier@6.11.0
+ prettier@1.19.1
+ eslint@4.19.1
+ eslint-plugin-prettier@3.1.3
+ eslint-plugin-joyent@2.1.0
added 104 packages from 136 contributors and audited 813 packages in 11.435s

20 packages are looking for funding
  run `npm fund` for details

found 5 vulnerabilities (3 low, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
/mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/node node_modules/.bin/eslint  bin/garbage-deleter.js lib/common.js lib/garbage-deleter.js test/mpu.test.js test/mpu-validation.test.js test/api.test.js test/mpu-concur.test.js test/common.js test/gc/garbage-deleter.slow.test.js test/mpu-large.test.js
bash -n bin/manta-mako-adm
bash -n tools/ngx_symcheck
/mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/node deps/eng/tools/bashstyle bin/manta-mako-adm
/mnt/tank/projects/tools-2461/manta-mako.git/build/node/bin/node deps/eng/tools/bashstyle tools/ngx_symcheck
deps/eng/tools/check-copyright -q
check ok
To run tests, run:

    export PATH=./build/node/bin:$PATH
    node $(find test/ -type f -name "*.js")

from the /opt/smartdc/mako directory on a storage instance.

Note that the 'tape' node module is required to run the mpu
tests, so you should rsync the updated node_modules directory
from your build zone into your storage zone before running the
tests.
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461) du -hs node_modules
40M     node_modules
timf@jenkins-agent-x86_64-19.4.0 (prr-TOOLS-2461)
```

Reply:
Right, and kthxbai reduces that size a bit further.